### ISSUE TITLE
[Suggestions/Improvements] Increasing information level for end-user

### ISSUE THREAD

**Issue by BlackTurtle123** [jansenmarc/WavesGatewayFramework#18]
Some suggestions I have thought about:
-Show both chains what block height was scanned last
-Use Waves as a variable in the static interface
-Show total fees.
-Be able to set a minimal treshold of tx in total with fees. This to prevent dust tx. Or unneccassary small spam tx that could fill up the tx list pool from the gateway.

**Comment by henning96** [comment]
Hello @BlackTurtle123, feel free to create a Pull Request.

**Comment by BlackTurtle123** [comment]
Lemme first find out how this Angular stuff works :D 
I never made a python app with angular, I never made a real angular app.
But yes maybee I will think about doing this

**Comment by BlackTurtle123** [comment]
@henning96 I am trying to edit the code.
However when I build the static pages, it seems it didn't picked up the python changes.
I tryed a local framework install but that didnt seemed to do the job either. Any suggestion? 

`ERROR in web\app\app.component.html(55,63): : Property 'gateway_fee' does not exist on type 'PublicConfiguration'.
web\app\app.component.html(63,63): : Property 'coin_fee' does not exist on type 'PublicConfiguration'.
`

**Comment by henning96** [comment]
Hello @BlackTurtle123, be sure to use -U option when doing local pip install, so that the changes are applied. There is also an editable mode option that makes a symlink to your WavesGatewayFramework directory (-e). You should use `ng serve` for local Angular development.

**Comment by BlackTurtle123** [comment]
I have used the -U flag. But it seems it just doesn't compile the python changes. While I am able to compile static files without python changes. So I am missing a command I guess to compile those new python files?

**Comment by henning96** [comment]
I don't understand. You don't compile Python?! So, there is no such command. It should be sufficient to sync the file changes with: `pip install -U .` in the WavesGatewayFramework directory.

**Comment by BlackTurtle123** [comment]
Mmmm yes that's what I did. Let me try again.
Or maybee the issue is I didn't edit enough files?

**Comment by BlackTurtle123** [comment]
What I mean is that I maybee had to edit more file to expose the variables beside the 1 I did and this the code just can't work.

**Comment by henning96** [comment]
What kind of variable do you mean? And where is it defined?

**Comment by BlackTurtle123** [comment]
`"""
PublicConfiguration
"""

from typing import Optional, Union
from decimal import Decimal


class PublicConfiguration:
    """
    Part of the configuration that is allowed to be available to the public,
    for example via ReST.
    """

    CUSTOM_CURRENCY_NAME_DICT_KEY = 'custom_currency_name'
    GATEWAY_WAVES_ADDRESS_DICT_KEY = 'gateway_waves_address'
    GATEWAY_COIN_HOLDER_DICT_KEY = 'gateway_coin_holder'
    WAVES_NODE_DICT_KEY = 'waves_node'
    WAVES_ASSET_ID = 'waves_asset_id'
    COIN_TRANSACTION_WEB_LINK = 'coin_transaction_web_link'
    WAVES_TRANSACTION_WEB_LINK = 'waves_transaction_web_link'
    COIN_ADDRESS_WEB_LINK = 'coin_address_web_link'
    WAVES_ADDRESS_WEB_LINK = 'waves_address_web_link'
    WEB_PRIMARY_COLOR = 'web_primary_color'
    GATEWAY_FEE = 'gateway_fee'
    COIN_FEE = 'coin_fee'

    def __init__(self,
                 custom_currency_name: str,
                 gateway_waves_address: str,
                 gateway_coin_address: str,
                 waves_node: str,
                 waves_asset_id: str,
                 waves_transaction_web_link: str,
                 waves_address_web_link: str,
                 coin_transaction_web_link: Optional[str] = None,
                 coin_address_web_link: Optional[str] = None,
                 web_primary_color: Optional[str] = None,
                 coin_fee : Optional[Union[int, Decimal]] = None,  # type
                 gateway_fee : Optional[Union[int, Decimal]]= None
                ) -> None:
        self._custom_currency_name = custom_currency_name
        self._gateway_waves_address = gateway_waves_address
        self._gateway_coin_holder = gateway_coin_address
        self._waves_node = waves_node
        self._waves_asset_id = waves_asset_id
        self._coin_transaction_web_link = coin_transaction_web_link
        self._waves_transaction_web_link = waves_transaction_web_link
        self._waves_address_web_link = waves_address_web_link
        self._coin_address_web_link = coin_address_web_link
        self._web_primary_color = web_primary_color
        self._coin_fee = coin_fee
        self._gateway_fee = gateway_fee

    @property
    def custom_currency_name(self) -> str:
        return self._custom_currency_name

    @property
    def gateway_waves_address(self) -> str:
        return self._gateway_waves_address

    @property
    def gateway_coin_holder(self) -> str:
        return self._gateway_coin_holder

    @property
    def waves_node(self) -> str:
        return self._waves_node

    @property
    def waves_asset_id(self) -> str:
        return self._waves_asset_id

    @property
    def waves_transaction_web_link(self) -> str:
        return self._waves_transaction_web_link

    @property
    def coin_address_web_link(self) -> Optional[str]:
        return self._coin_address_web_link

    @property
    def coin_transaction_web_link(self) -> Optional[str]:
        return self._coin_transaction_web_link

    @property
    def waves_address_web_link(self) -> str:
        return self._waves_address_web_link

    @property
    def web_primary_color(self) -> str:
        return self._web_primary_color

    @property
    def coin_fee(self) -> Decimal:
        return self._coin_fee

    @property
    def gateway_fee(self) -> Decimal:
        return self._gateway_fee
`

**Comment by BlackTurtle123** [comment]
I tryed to expose the gateway_fee and the coin_fee.
Above is the file I edited.

**Comment by henning96** [comment]
You have to reexport those variables in the __init__.py file if you want to use them outside of the package.

**Comment by henning96** [comment]
It is done in any `__init__.py` file in the project. So, pick one.

**Comment by BlackTurtle123** [comment]
But what I don't get, why are the variables that you have used never exposed in an __init__.py? It seems I would need to do the same flow? That's why I was wondering in which init you were exposing those variables in particular.

Actually I am totally lost in the variable flow how to expose them. Since I did add a few in a file that is default exposed and made edits on the locations I saw the other variables.

**Comment by BlackTurtle123** [comment]
I tryed to start with something simple, add a variable for the WAVES, however, since 3 tests fail something goes wrong and can't figured out what.
https://github.com/jansenmarc/WavesGatewayFramework/pull/19
If you would have time to help me find the issue/ what I forgot/ did wrong it would be amazing.

**Comment by henning96** [comment]
On my machine only one test fails and that is because you are overhanding an unexpected parameter to a constructor function?

**Comment by BlackTurtle123** [comment]
Yeah I was trying to instead have WAVES fixed everywhere to make it dynamic, so it's easier and clean to replace for forks from waves, in my case Turtle Network. However not sure where I made the ruin since I thought I extended it everyhere

**Comment by henning96** [comment]
Closed, because nothing happened here since nearly one year.
